### PR TITLE
Issue#47: Install and configure sass npm package

### LINKED ISSUE #47: Install and Configure SCSS Compilation Using the sass npm Package
## Summary

We need to add SCSS support to the project to improve the maintainability and organization of styles across components. SCSS offers enhanced styling capabilities, such as variables, nesting, and mixins, which will help make the code more modular and scalable.

## Acceptance Criteria

- [x]   Install the sass npm package globally via `npm install -g sass.`
- [x]  Ensure SCSS files are imported and used in TypeScript components.
- [x]  Update at least one component to demonstrate the use of SCSS.

### PR DESCRIPTION
**Pull request by zahrafalak**
## Description

We need to add SCSS support to the project to improve the maintainability and organization of styles across components. SCSS offers enhanced styling capabilities, such as variables, nesting, and mixins, which will help make the code more modular and scalable.

## Related Issue

Closes #47 

## Acceptance Criteria

- [x] Install the sass npm package globally via `npm install -g sass`.
- [x] Ensure SCSS files are imported and used in TypeScript components.
- [x] Update at least one component to demonstrate the use of SCSS.

## Type of Changes

`enhancement`  `feature`

## Notes

- The AuthenticatedNavBar.scss file is updated to demonstrate the use of scss. The nested CSS selectors show a clear visual hierarchy. 
- A new partial `_variables.scss` is added under `styles` folder with some placeholder values for colours. 

[Related document resource](https://sass-lang.com/guide/)

### CODE REVIEW

**Review by alex-andria — APPROVED** [review]
I saw you wrote an example inside the acceptance criteria. But approving for now in the meantime!

File: package.json [hunk 1]
```diff
@@ -34,6 +34,7 @@
 		"jsdom": "^24.1.1",
 		"lint-staged": "^15.2.9",
 		"prettier": "^3.3.3",
+		"sass-embedded": "^1.79.3",
```
Comment (bbland1) [inline package.json:37]: question: when running npm start i am getting an error about `sass` not installed after running `npm i` when switching to the branch. i'm not seeing it in the `devDependenceies` and i just want to check if it is something i am doing wrong 😅

File: src/components/AuthenticatedNavBar.tsx [hunk 2]
```diff
@@ -2,22 +2,22 @@ import React from "react";
...
 import { SignOutButton } from "../api";
 
-import "./AuthenticatedNavBar.css";
+import "./AuthenticatedNavBar.scss";
 
 export function AuthenticatedNavBar() {
 	return (
 		<nav className="Nav">
-			<div className="Nav-container">
+			<div className="Nav__container">
```
Comment (bbland1) [inline src/components/AuthenticatedNavBar.tsx:10]: praise: thanks for the example! still being new to it i was lost for a moment but i think i'm starting to see the vision 😂.

**Comment by bbland1** [comment]
Thanks for doing this! I used the preview link to just see it was all still working and looking at it and it was well done. I really appreciate the example on the `AuthenticatedNavBar`! I've done some reading about `bootstrap` and `sass` so I think it is going to be really cool to have both of them! Seeing the changes in the `className` and how that will integrate with the `bootstrap` components is going to be quite nice. I'm excited to get the colors in there and see the process flow.

File: package.json (hunk 1 above)
Comment (zahrafalak) [reply package.json:37]: Huh, interesting. The error doesn't show up on my end so not sure what's up. Did you try to see if it works in the preview? If it works there and doesn't work when you checkout the branch, maybe try deleting the node modules and do `npm ci`?

**Comment by zahrafalak** [comment]
> Thanks for doing this! I used the preview link to just see it was all still working and looking at it and it was well done. I really appreciate the example on the `AuthenticatedNavBar`! I've done some reading about `bootstrap` and `sass` so I think it is going to be really cool to have both of them! Seeing the changes in the `className` and how that will integrate with the `bootstrap` components is going to be quite nice. I'm excited to get the colors in there and see the process flow.

Good to know that it was still working in preview 😄 Thanks for the feedback! Yeah, I do think this will work great together with the bootstrap components.

**Pull request merged**